### PR TITLE
Add onboarding finalization for users after successful payment

### PR DESCRIPTION
**Pull request by RIDUMATICS**
Add onboarding finalization for attendees after successful payment

- Implement endpoint to collect bio data from registered users
- Add validation for bio data submission
- Update users model to include necessary bio data fields

### CODE REVIEW

File: src/http/controllers/users/user.dto.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  UserExpertiseLevel,
+  UserRole,
+  UserShirtSize,
+} from "../../../users/user.model";
+
+const isRequiredString = () => joi.string().required();
+
+export const saveUserBioDataDTO = joi.object().keys({
+  fullname: isRequiredString(),
+  email_address: isRequiredString().email({ tlds: { allow: false } }),
```
Comment (Emmanuerl) [inline src/http/controllers/users/user.dto.ts:12]: why the TLD strictness on the email ?

File: src/users/user.model.ts [hunk 2]
```diff
@@ -1,14 +1,36 @@
 import { Model } from "../internal/postgres";
 
+export enum UserRole {
```
Comment (Emmanuerl) [inline src/users/user.model.ts:3]: typed strings over enums, enums are generally not the best way to go.

File: src/users/user.model.ts [hunk 3]
```diff
@@ -1,14 +1,36 @@
...
+  SOFTWARE_ENGINEER = "Software Engineer",
+  PRODUCT_DESIGNER = "Product Designer",
+  PRODUCT_MANAGER = "Product Manager",
+  QA_ANALYST = "QA Analyst",
+  TECHNICAL_WRITER = "Technical Writer",
+  OTHERS = "Others",
+}
+
+export enum UserExpertiseLevel {
```
Comment (Emmanuerl) [inline src/users/user.model.ts:12]: This is what i expect instead of enums
```ts
export const expertise = <const>["beginner","intermediate","expert"];
export type Expertise = typeof expertise[number]
```
see [this line](https://github.com/GDG-W/Asgard/blob/3e1c1277948e240dd2bb0ba9bb591318d7d90727/src/tickets/ticket.model.ts#L41)

File: src/http/controllers/users/user.controller.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import { controller } from "../../../internal/http";
+import { User, UserAlreadyExist } from "../../../users";
+import { userService } from "../../../users/user.service";
+import { ApplicationError } from "../../../internal/error";
+import { StatusCodes } from "http-status-codes";
+
+async function saveUserBioData(req: Request, res: Response) {
+  try {
+    const body: User = req.body;
+    body.ticket_id = "054f23b0-8687-49e1-92a1-240ad8c6d6ba"; //TODO: To be picked from loginUser credentials
```
Comment (Emmanuerl) [inline src/http/controllers/users/user.controller.ts:13]: This is incorrect, the ticket represents the type of ticket they acquired, so the user's `id`,`email_address` and `ticket_id` will already be populated before users get to finalise their onboarding.

tldr: remove this line

File: src/users/user.service.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    const trx = await knex.transaction();
+    try {
+      // check if a user exist with the ticket_id
+      const existingUser = await this.userRepo.findOne(trx, {
+        ticket_id: userDTO.ticket_id,
+      });
+
+      if (existingUser) throw new UserAlreadyExist(existingUser.ticket_id);
+
+      const user = await this.userRepo.create(trx, {
```
Comment (Emmanuerl) [inline src/users/user.service.ts:18]: no, we're not creating users here, it's supposed to be an update.

**Review by Emmanuerl — CHANGES_REQUESTED** [review]
@RIDUMATICS it seems you didn't properly understand what i was asking of you, let's sync ASAP so i can explain things better
